bin/varnishd/cache/cache_backend.c

```diff
@@ -105,6 +111,38 @@ VBE_Connect_Error(struct VSC_vbe *vsc, int err)
 			dst = cache_param->tmx;				\
 	} while (0)
 
+#define BE_BUSY(be)	\
+	(be->max_connections > 0 && be->n_conn >= be->max_connections)
+
+/*--------------------------------------------------------------------*/
+
+static void
+vbe_connwait_signal_locked(struct backend *bp)
+{
+	struct busyobj *bo;
+
+	Lck_AssertHeld(bp->director->mtx);
+
+	if (bp->n_conn < bp->max_connections) {
+		bo = VTAILQ_FIRST(&bp->cw_head);
+		if (bo != NULL) {
+			CHECK_OBJ(bo, BUSYOBJ_MAGIC);
+			AZ(pthread_cond_signal(&bo->cw_cond));
+		}
+	}
+}
+
+static void
+vbe_connwait_dequeue(struct backend *bp, struct busyobj *bo, int lock_it)
+{
+	if (lock_it)
```

Comment:
um no to `lock_it`.
Can we please have a `_locked` function, or just do the `VTAILQ_REMOVE()` wherever we want the locked variant?

Reply:
Originally the dequeue was performed at each location, doing what was needed at that spot.  But questions were raised about duplicating code, which led to this helper. I'd prefer it just go back inline where needed.

Reply:
Removed `vbe_connwait_dequeue` and replaced it with inlined calls

Reply:
see top level comment